sycl/source/detail/graph_impl.hpp

```diff
@@ -476,6 +426,29 @@ class graph_impl {
     }
   }
 
+  /// Recursively check successors of NodeA and NodeB to check they are similar.
+  /// @param NodeA pointer to the first node for comparison
+  /// @param NodeB pointer to the second node for comparison
+  /// @return true is same structure found, false otherwise
+  bool checkNodeRecursive(std::shared_ptr<node_impl> NodeA,
```

Comment:
Should this function be `static`? If i'm reading this modification right, it shouldn't need `this`

Reply:
I've made it static 👍